graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultInputObjectMapper.kt

```diff
@@ -16,44 +16,64 @@
 
 package com.netflix.graphql.dgs.internal
 
-import com.fasterxml.jackson.module.kotlin.isKotlinClass
 import com.netflix.graphql.dgs.exceptions.DgsInvalidInputArgumentException
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import org.springframework.core.KotlinDetector
+import org.springframework.util.CollectionUtils
 import org.springframework.util.ReflectionUtils
-import java.lang.reflect.*
+import java.lang.reflect.Field
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+import java.lang.reflect.TypeVariable
+import java.lang.reflect.WildcardType
 import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
 import kotlin.reflect.KType
 import kotlin.reflect.full.primaryConstructor
 import kotlin.reflect.jvm.javaType
 import kotlin.reflect.jvm.jvmErasure
 
 @Suppress("UNCHECKED_CAST")
-class DefaultInputObjectMapper(val customInputObjectMapper: InputObjectMapper? = null) : InputObjectMapper {
+class DefaultInputObjectMapper(private val customInputObjectMapper: InputObjectMapper? = null) : InputObjectMapper {
     private val logger: Logger = LoggerFactory.getLogger(InputObjectMapper::class.java)
 
     override fun <T : Any> mapToKotlinObject(inputMap: Map<String, *>, targetClass: KClass<T>): T {
-        val params = targetClass.primaryConstructor!!.parameters
-        val inputValues = mutableListOf<Any?>()
+        val constructor = targetClass.primaryConstructor
+            ?: throw DgsInvalidInputArgumentException("No primary constructor found for class $targetClass")
+
+        val parameters = constructor.parameters
+        val parametersByName = CollectionUtils.newLinkedHashMap<KParameter, Any?>(parameters.size)
+
+        for (parameter in parameters) {
+            if (parameter.name !in inputMap) {
+                if (parameter.isOptional) {
+                    continue
+                } else if (parameter.type.isMarkedNullable) {
+                    parametersByName[parameter] = null
```

Comment:
This preserves the existing behavior (as exercised in the tests), but I am not sure this is desirable. It essentially means that we fill in nulls for classes like this when the argument isn't explicitly supplied:

```data class Foo(val nullableParam: String?)```

I would think that if that were the desired behavior, the user would write the class like:

```data class Foo(val nullableParam: String? = null)```

Anyway, I've left it here so as to not break the existing behavior.

Reply:
Yeah, I think this has been a point of contention in the codegen plugin as well. But I agree with the decision to preserve existing behavior.